### PR TITLE
dex.trades: add dodoex bsc

### PR DESCRIPTION
**Pull request by yongjun925**
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] the directory tree matches the pattern /sector/blockchain/ e.g. /tokens/ethereum
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each file has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

### CODE REVIEW

**Comment by yongjun925** [comment]
@milkyklim Hello milkyklim, could you help me to review this pr for add dodoex to dex.trades.

**Comment by jeff-dude** [comment]
thank you @yongjun925 for the PR. we are in the process of getting abstractions in bsc database back up to speed after some downtime. we can look to get this implemented next week ideally.

**Comment by jeff-dude** [comment]
@yongjun925 @milkyklim we are ready to implement this when you finalize your review

**Comment by yongjun925** [comment]
@jeff-dude  I have tested and verified it. Can you help me merge it. thanks!
 https://dune.com/embeds/923436/1614720/f37558e4-fff3-4cb3-ada2-689bb7e298b8

**Comment by yongjun925** [comment]
@jeff-dude What are the remaining issues with this merge request. Thanks.

**Comment by jeff-dude** [comment]
> @jeff-dude What are the remaining issues with this merge request. Thanks.

no issues at this point -- i'll start to build it now

File: binance_smart_chain/dex/trades/insert_dodo.sql [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+            dodoex."DSP_evt_DODOSwap"
+        WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+    ) dexs
+    INNER JOIN bsc.transactions tx
+        ON dexs.tx_hash = tx.hash
+        AND tx.block_time >= start_ts
+        AND tx.block_time < end_ts
+        AND tx.block_number >= start_block
+        AND tx.block_number < end_block
+    LEFT JOIN bep20.tokens erc20a ON erc20a.contract_address = dexs.token_a_address
```
Comment (jeff-dude) [inline binance_smart_chain/dex/trades/insert_dodo.sql:246]: when you join this table, it's used to pull token symbol and use the decimal values for calculating raw token amount. i've noticed that the dodoex decoded tables have a distinct count of token addresses which is greater than the number found after the join. this causes null values for token symbols and raw amounts in end result. you may need to add all distinct token addresses (both a and b) from the decoded tables to this file:
https://github.com/duneanalytics/abstractions/blob/master/binance_smart_chain/bep20/tokens.sql

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: binance_smart_chain/dex/trades/insert_dodo.sql (hunk 1 above)
Comment (jeff-dude) [reply binance_smart_chain/dex/trades/insert_dodo.sql:246]: reference query:
https://dune.com/queries/956021

switch between left and inner join helps show difference

Comment (yongjun925) [reply binance_smart_chain/dex/trades/insert_dodo.sql:246]: I deleted some of the same symbols. the contract_address should not be duplicate entries.

Comment (yongjun925) [reply binance_smart_chain/dex/trades/insert_dodo.sql:246]: Sorry, I didn't notice the case. Now I've updated it again.

Comment (jeff-dude) [reply binance_smart_chain/dex/trades/insert_dodo.sql:246]: looks good!

**Review by jeff-dude — CHANGES_REQUESTED** [review]
@yongjun925

File: binance_smart_chain/dex/trades/insert_dodo.sql [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    ON CONFLICT DO NOTHING
+    RETURNING 1
+)
+SELECT count(*) INTO r from rows;
+RETURN r;
+END
+$function$;
+
+-- fill 2020
+SELECT dex.insert_dodo(
```
Comment (jeff-dude) [inline binance_smart_chain/dex/trades/insert_dodo.sql:275]: please edit function call to correct name: dex.insert_dodoex_bsc

this applies to all historical calls below -- cron job seems to be correct

**Review by jeff-dude — CHANGES_REQUESTED** [review]
@yongjun925

**Comment by jeff-dude** [comment]
the code is deployed and data should be loaded, once two final changes are in place, we can merge the PR

**Comment by yongjun925** [comment]
@jeff-dude OK, it has been changed. Thank you for your guidance

File: binance_smart_chain/dex/trades/insert_dodo.sql [hunk 3]
```diff
@@ -236,8 +250,10 @@ WITH rows AS (
         FROM
             dodoex."DSP_evt_DODOSwap"
         WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+            AND evt_block_time >= start_ts
+            AND evt_block_time < end_ts
     ) dexs
-    INNER JOIN bsc.transactions tx
+    LEFT JOIN bsc.transactions tx
```
Comment (jeff-dude) [inline binance_smart_chain/dex/trades/insert_dodo.sql:256]: any reason this changed to LEFT join? the code ran as inner join, is there an impact on row counts and null values in select columns with LEFT used?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: binance_smart_chain/dex/trades/insert_dodo.sql (hunk 3 above)
Comment (yongjun925) [reply binance_smart_chain/dex/trades/insert_dodo.sql]: error occurred during verification, then i change to back, please check it for me , thanks

Comment (jeff-dude) [reply binance_smart_chain/dex/trades/insert_dodo.sql]: yep, looks good, thank you for the patience and working with me to get this integrated!

**Comment by jeff-dude** [comment]
due to merge conflicts & no access to your forked repo, i've opened a new PR with the same changes and will merge that one in, while closing this PR.
reference: #1233

**Pull request closed without merging**